btb/benchmark/__init__.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+
+import pandas as pd
+
+from btb.benchmark.challenges import Bohachevsky, Branin, Rosenbrock
+
+DEFAULT_CHALLENGES = [
+    Bohachevsky,
+    Branin,
+    Rosenbrock,
+]
+
+
+def benchmark(tuning_functions, challenges=DEFAULT_CHALLENGES, iterations=1000):
+    """Benchmark function.
+
+    This benchmark function iterates over a collection of ``challenges`` and executes a
+    ``tuner_function`` for each one of the ``challenges`` for a given amount of iterations.
+
+    Args:
+        tuning_functions (dict):
+            Python dictionary with the ``name`` of the function as ``key`` and the function that
+            returns the best score for a given ``scorer``. This function must have three arguments:
+
+                * scorer (function):
+                    A function that performs scoring over params.
+
+                * tunable (btb.tuning.Tunable):
+                    A ``Tunable`` instance used to instantiate a tuner.
+
+                * iterations (int):
+                    Number of tuning iterations to perform.
+
+        challenges (single challenge or list):
+            A single ``challenge`` or a list of ``chalenges``. This challenges must inherit
+            from ``btb.challenges.challenge.Challenge``.
+        iterations (int):
+            Amount of iterations to perform for the ``tuner_function``.
+
+    Returns:
+        pandas.DataFrame:
+            A ``pandas.DataFrame`` with the obtained scores for the given challenges is being
+            returned.
+    """
+
+    if not isinstance(challenges, list):
+        challenges = [challenges]
+
+    results = list()
+
+    for challenge_class in challenges:
+        challenge = challenge_class()
+        tunable = challenge.get_tunable()
+        for name, function in tuning_functions.items():
+            score = function(challenge.evaluate, tunable, iterations)
+
+            result = pd.Series({
```

Comment:
This generates a table like this:

|   | challenge  | tuner            | score | iterations | avg  |
|---|------------|------------------|-------|------------|------|
| 0 | Rosenbrock | my_function      | 0.9   | 1000       | 0.75 |
| 1 | Branin     | my_function      | 0.8   | 1000       | 0.75 |
| 2 | Rosenbrock | another_function | 0.7   | 1000       | 0.75 |
| 3 | Branin     | another_function | 0.6   | 1000       | 0.75 |

Which is not that useful, especially because the iterations value is already known, and the avg has no meaning, as it has been computed across all the rows.

Let's change it to this format instead:

|                  | Rosenbrock | Branin | Mean | Std  |
|------------------|------------|--------|------|------|
| my_function      | 0.9        | 0.8    | 0.85 | 0.05 |
| another_function | 0.7        | 0.6    | 0.75 | 0.05 |